src/SourceGenerators/Uno.UI.SourceGenerators/HotRestart/HotRestartGenerator.cs

```diff
@@ -0,0 +1,126 @@
+﻿#nullable enable
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Uno.UI.SourceGenerators.Helpers;
+using System.Diagnostics;
+using Uno.Extensions;
+using Uno.Roslyn;
+
+namespace Uno.UI.SourceGenerators.HotRestart;
+
+[Generator]
+public partial class HotRestartGenerator : ISourceGenerator
+{
+	public void Initialize(GeneratorInitializationContext context)
+	{
+	}
+
+	public void Execute(GeneratorExecutionContext context)
+	{
+		if (!DesignTimeHelper.IsDesignTime(context)
+			&& PlatformHelper.IsIOS(context)
+			&& PlatformHelper.IsUnoHead(context))
+		{
+			var generator = new Generator(context);
+			generator.Run();
+		}
+	}
+
+	private class Generator
+	{
+		private readonly GeneratorExecutionContext _context;
+
+		public Generator(GeneratorExecutionContext context)
+		{
+			_context = context;
+		}
+
+		public void Run()
+		{
+			if (!IsGenerationEnabled())
+			{
+				return;
+			}
+
+			var mauiAppDelegate = _context.Compilation.GetTypeByMetadataName("Microsoft.Maui.MauiUIApplicationDelegate");
+			var shouldDefineAppDelegate = (mauiAppDelegate is null).ToString().ToLowerInvariant();
+
+			var appType = GetApplicationDefinitionType();
+
+			if (appType is null)
+			{
+				return;
+			}
+
+			var document = $$"""
+				#if DEBUG
+				// <auto-generated>
+				// *************************************************************
+				// This file has been generated by Uno.UI ({{nameof(HotRestartGenerator)}})
+				// *************************************************************
+				// </auto-generated>
+
+				#if {{shouldDefineAppDelegate}}
+				namespace Microsoft.Maui
+				{
+					[global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]
+					internal class MauiUIApplicationDelegate : global::UIKit.UIApplicationDelegate
+					{
+					}
+				}
+				#endif
+
+				[global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]
+				internal class __UnoHotRestartDelegate : global::Microsoft.Maui.MauiUIApplicationDelegate
+				{
+					public override bool FinishedLaunching(global::UIKit.UIApplication uiApplication, global::Foundation.NSDictionary launchOptions)
+					{
+						#if HAS_UNO_WINUI
+						Microsoft.UI.Xaml.Application.Start(_ => new {{appType}}());
+						#else
+						Windows.UI.Xaml.Application.Start(_ => new {{appType}}());
+						#endif
```

Comment:
- Consider prefixing with `global::`
- Not sure, but I thought `Windows.UI.Xaml` would get replaced by `Microsoft.UI.Xaml` when the conversion happens. So maybe we can just add a single line that uses `Windows.UI.Xaml`?

Reply:
Indeed, it should be renamed automatically!